src/browser/browser.js

```diff
@@ -160,9 +165,16 @@ define((require, exports, module) => {
 
     const theme = Browser.readTheme(activeWebViewerCursor);
 
+    const hotZone = immutableState.get('hotZone');
+
+    if (isTabStripVisible) {
+      showNavbar();
+    }
```

Comment:
React really advices against touching DOM during render phase. I think it would be much better if you could handle DOM updates in one of the virtual attributes like you already do with `mouseInHotZone`. In fact I don't really understand why it can't be something like:

```js
Main({
  navbarVisible: isTabStripVisible || hotZone.get('mouseIn'),
  //...
})
```

That would make more cleaner and also would not go against React's don't touch DOM during render phase.